### PR TITLE
Sockets

### PR DESCRIPTION
**Pull request by TriauNicolas**
# Main changes
- Add CRUD sockets Table / Field / Edges
- Type related to events

## Others
- New way of triggering the API

### CODE REVIEW

File: src/hooks/useEdges.ts [hunk 1]
```diff
@@ -18,20 +20,67 @@ export const useEdges = () => {
 
   const onEdgeUpdate = useCallback((oldEdge: Edge, newConnection: Connection) => {
     edgeUpdateSuccessful.current = true;
+
+    console.log(oldEdge);
+    console.log(newConnection);
```
Comment (gabrielseron) [inline src/hooks/useEdges.ts]: delete console logs

File: src/hooks/useEdges.ts [hunk 2]
```diff
@@ -18,20 +20,67 @@ export const useEdges = () => {
...
     edgeUpdateSuccessful.current = true;
-  }, []);
+  }, [getNode]);
 
   const onConnect: OnConnect = useCallback(
-    (connection) => setEdges((eds) => addEdge(connection, eds)),
-    [setEdges]
+    (connection) => {
+
+      console.log(connection);
```
Comment (gabrielseron) [inline src/hooks/useEdges.ts]: logs

File: src/sockets/socketConnection.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    transportOptions: {
+      polling: {
+        extraHeaders: {
+          BearerToken: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImZhNjliOWU3LWMwMGItNDM4Mi1iNTExLTBlM2Q5OGRkZTgyNiIsImlhdCI6MTY5NzI1MjE5MCwiZXhwIjoxNjk3ODU2OTkwfQ.aZdtn066i30BWFW0H6Lw7fJbnk9za4uA5zh8iIue2GQ'
+        }
+      }
+    }
+  });
+  
+  console.log(socket)
```
Comment (gabrielseron) [inline src/sockets/socketConnection.ts]: log

File: src/hooks/useDataToJson.ts [hunk 4]
```diff
@@ -25,6 +24,7 @@ export const useDataToJson = ({ nodes, edges }: DataToJsonProps) => {
       
       const fieldsTable = nodes.filter((node: Node) => table.id == node.parentNode)
       fieldsTable.forEach((node: Node) => {
+        console.log(node);
```
Comment (gabrielseron) [inline src/hooks/useDataToJson.ts]: log

**Review by gabrielseron — CHANGES_REQUESTED** [review]

File: src/components/FieldModal/FieldModal.tsx [hunk 5]
```diff
@@ -56,55 +56,33 @@ export const FieldModal = ({ idTable, closeModal, idField }: FieldModalProps) =>
...
       }
     } 
     else {
-      const currentNode: any = idField ? getNode(idField) : undefined;
+      const currentNode: Node<any> | undefined = idField ? getNode(idField) : undefined;
 
       if (currentNode) {
-        const allNodes = getNodes();
-        const titleParent = (infosField?.data.title).split('.')[0];
+        // const allNodes = getNodes();
```
Comment (valentinsejourne) [inline src/components/FieldModal/FieldModal.tsx]: comm utile ?

**Review by gabrielseron — APPROVED** [review]

File: src/hooks/useNodes.ts [hunk 6]
```diff
@@ -15,15 +15,15 @@ const basicStyleTableNode: {} = {
...
-  { id: '4', type: 'fieldNode', position: { x: 0, y: 50 }, positionAbsolute: { x: 0, y: 50 }, data: { title: "UsersGroup.name", name: 'name', type: 'varchar(20)' }, parentNode: '3', draggable: false, selected: false },
-];
+// const initialNodes = [
+//   { id: '1', type: 'tableNode', position: { x: 0, y: 0 }, positionAbsolute: { x: 0, y: 0 }, data: { title: "Users" }, style: basicStyleTableNode, expandParent: true, selected: false, draggable: true },
+//   { id: '2', type: 'fieldNode', position: { x: 0, y: 50 }, positionAbsolute: { x: 0, y: 50 }, data: { title: "Users.name", name: 'name', type: 'varchar(20)' }, parentNode: '1', draggable: false, selected: false },
+//   { id: '3', type: 'tableNode', position: { x: -400, y: 50 }, positionAbsolute: { x: 0, y: 0 }, data: { title: "UsersGroup" }, style: basicStyleTableNode, expandParent: true, selected: false, draggable: true },
+//   { id: '4', type: 'fieldNode', position: { x: 0, y: 50 }, positionAbsolute: { x: 0, y: 50 }, data: { title: "UsersGroup.name", name: 'name', type: 'varchar(20)' }, parentNode: '3', draggable: false, selected: false },
+// ];
```
Comment (valentinsejourne) [inline src/hooks/useNodes.ts:23]: same

**Review by valentinsejourne — APPROVED** [review]

File: src/hooks/useNodes.ts (hunk 6 above)
Comment (TriauNicolas) [reply src/hooks/useNodes.ts:23]: Test en local sans socket

**Pull request merged**